incubator/hnc/internal/validators/hierarchy_test.go

```diff
@@ -96,6 +98,54 @@ func TestChangeParentOnManagedBy(t *testing.T) {
 	}
 }
 
+func TestChangeParentWithConflict(t *testing.T) {
+	f := foresttest.Create("-a-c") // a <- b; c <- d
+
+	// Set secret to "Propagate" mode.
+	or := &reconcilers.ObjectReconciler{
+		GVK:  schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Secret"},
+		Mode: api.Propagate,
+	}
+	f.AddTypeSyncer(or)
+
+	// Create secrets with the same name in namespace 'a' and 'd'.
+	createSecret("conflict", "a", f)
+	createSecret("conflict", "d", f)
+
+	h := &Hierarchy{Forest: f}
+	l := zap.Logger(false)
+
+	tests := []struct {
+		name string
+		nnm  string
+		pnm  string
+		fail bool
+	}{
+		{name: "conflict in subtree root and new parent", nnm: "a", pnm: "d", fail: true},
+		{name: "conflict in itself and new parent", nnm: "d", pnm: "a", fail: true},
```

Comment:
What's the difference between these? Just that `a` is a root and `d` is a leaf?

Reply:
Ah right, changed the new parent in the 2nd case to `b` and updated the first test name:
```
{name: "conflict in itself and the new parent", nnm: "a", pnm: "d", fail: true},
{name: "conflict in itself and a new ancestor (not the parent)", nnm: "d", pnm: "b", fail: true},
```
So that the 4 failed tests covers all combinations of conflicts between`itself` and `parent/ancestor`, `subtree leaf` and `parent/ancestor`.